### ISSUE TITLE
[doc] 6.1.3.1.5  Adding Grafana Dashboards - incomplete "Use our pre-built dashboards to monitor the SUSE CaaS Platform system"

### ISSUE THREAD

**Issue by Martin-Weiss** [SUSE/doc-caasp#842]
6.1.3.1.5  Adding Grafana Dashboards

https://susedoc.github.io/doc-caasp/master/single-html/caasp-admin/#adding_grafana_dashboards

-----

Following this example - we do not have all dashboards and the etcd one does not work because there is further adjustment required.

Maybe we should "copy" https://github.com/SUSE/caasp-monitoring/blob/master/README.md or just link in this git project?

**Comment by jenting** [comment]
I'd prefer to link the GitHub repo directly since we might add more pre-built dashboards in the future.

**Comment by innobead** [comment]
> I'd prefer to link the GitHub repo directly since we might add more pre-built dashboards in the future.

1. Yes, the easy way is to link to the repo but I found the README of the repo is not complete, and it just describes well about `etcd cluster monitoring` part only. So it's necessary to improve.
2. Please also think further about one case: the release gap between CaaSP release and `caasp-monitoring` repo. I believe there is a delay gap between the current CaaSP release and `caasp-monitoring` master branch, so some code merged in `caasp-monitoring` does not mean it is ready in the current release or maybe ready in the following release.

So, basically I would suggest still add the instructions of installing builtin dashboards in the documentation, but for the detailed about the prerequisites needed for some specific dashboard like `etcd cluster`, we can just add a link to the repo:README.md to save duplicated effort. WDYT?

**Comment by Martin-Weiss** [comment]
The problems I see with linking to github:
- probably causing support problems as external stuff might not be supported
- QA might not be able to test and might not see changes on external sources
- not included in the PDF or other documentation downloads

**Comment by jenting** [comment]
I was thinking to make these SUSE provided grafana dashboard merged into SUSE release grafana helm chart. Then we have the version information for support and QA on which version customer reported or which version QA had verified. Would it be better? @innobead @Martin-Weiss 

But I have to make sure this idea works in the air-gapped environment.

**Comment by Martin-Weiss** [comment]
I believe we just would have to deliver the json / yaml files and the documentation on how to add own or SUSE or official available dashboards via configmaps.

I believe adding them direct into the helm chart makes it difficult to maintain and upgrade, adjust and customize and also might make it difficult for customers that already have a Grafana running "somewhere else".

Maybe we should just follow the official Grafana process and push our dashboards to grafana.com?

For air-gapped - we just need to create a yaml out of the json - so maybe a simple script that "migrates any given dashboard json" to a configmap yaml would already be sufficient for our customers?

**Comment by innobead** [comment]
> I believe we just would have to deliver the json / yaml files and the documentation on how to add own or SUSE or official available dashboards via configmaps.
> 
> I believe adding them direct into the helm chart makes it difficult to maintain and upgrade, adjust and customize and also might make it difficult for customers that already have a Grafana running "somewhere else".
> 
Hmm, it seems there is a need for portability for customers.

> Maybe we should just follow the official Grafana process and push our dashboards to grafana.com?
> 
This is a good suggestion. cc @flavio 

> For air-gapped - we just need to create a yaml out of the json - so maybe a simple script that "migrates any given dashboard json" to a configmap yaml would already be sufficient for our customers?

@jenting I would suggest focusing on fixing this issue first, then we can come out another ticket to tackle something for integration or some specific purpose.

**Comment by jenting** [comment]
> For air-gapped - we just need to create a yaml out of the json - so maybe a simple script that "migrates any given dashboard json" to a configmap yaml would already be sufficient for our customers?

This [repo](https://github.com/eedugon/grafana-dashboards-configmap-generator) might fulfill your requirement although the downloaded dashboard must be with suffixed `-dashboard.json` or -`datasource.json`, and the `--namespace` flag only works with `--apply-configmap` flag.

**Comment by Martin-Weiss** [comment]
> 
> 
> > For air-gapped - we just need to create a yaml out of the json - so maybe a simple script that "migrates any given dashboard json" to a configmap yaml would already be sufficient for our customers?
> 
> This repo [grafana-dashboards-configmap-generator](https://github.com/eedugon/grafana-dashboards-configmap-generator) might fulfill your requirement although the downloaded dashboard must be with suffixed `-dashboard.json` or -`datasource.json`, and the `--namespace` flag only works with `--apply-configmap` flag.
> 
> EDIT:
> I made some modifications to generates the namespace in ConfigMap. https://github.com/jenting/grafana-dashboards-configmap-generator

Yes - that looks like what we might need to deliver the dashboards! Maybe we can add this to our product delivery? I believe this really would give added value..

As a side-note.. we also need a solution documented to add Grafana plugins..

**Comment by flavio** [comment]
> > Maybe we should just follow the official Grafana process and push our dashboards to grafana.com?
> 
> This is a good suggestion. cc @flavio

I don't see any problem with doing that :rocket:

**Comment by jenting** [comment]
> > > Maybe we should just follow the official Grafana process and push our dashboards to grafana.com?
> > 
> > 
> > This is a good suggestion. cc @flavio
> 
> I don't see any problem with doing that

@flavio 
The next question is how do I contribute the dashboards to https://grafana.com/grafana/dashboards on behalf of SUSE? Now, I only have my personal grafana account.

**Comment by jenting** [comment]
Because we don't have a unique SUSE account to contribute and maintains our pre-built dashboard to https://grafana.com/grafana/dashboards, it's an organization corporation problem and more like a feature request.

I think this issue will be stale for a long time, I'd recommend creating a feature request in JIRA for this and close this issue. What are other people thinking?

**Comment by r0ckarong** [comment]
> Because we don't have a unique SUSE account to contribute and maintains our pre-built dashboard to https://grafana.com/grafana/dashboards, it's an organization corporation problem and more like a feature request.
> 
> I think this issue will be stale for a long time, I'd recommend creating a feature request in JIRA for this and close this issue. What are other people thinking?

Yes, please create a feature request.
